module/zfs/spa_misc.c

```diff
@@ -310,6 +317,16 @@ unsigned long  zfs_deadman_checktime_ms = 5000ULL;
  */
 int zfs_deadman_enabled = 1;
 
+/*
+ * Controls the behavior of the deadman when it detects a "hung" I/O.
+ * Valid values are zfs_deadman_failmode=<wait|continue|panic>.
+ *
+ * wait     - Wait for the "hung" I/O (default)
+ * continue - Attempt to recover from a "hung" I/O
+ * panic    - Panic the system
+ */
+char *zfs_deadman_failmode = "wait";
```

Comment:
Presumably the rationale for using "wait" as the default instead of "continue" (which would probably be more useful) is not to change the current behavior (much)?

Reply:
That and possible that the "continue" behavior might have side effects or result in additional instability.  We can revisit the default setting once we're more comfortable with the proposed recovery logic.  But for now let's keep the core behavior the same aside from improving the logging.